### ISSUE TITLE
dbGetQuery returns Error in new_result(connection@ptr, statement) : std::bad_alloc

### ISSUE THREAD

**Issue by mbfsdatascience** [r-dbi/odbc#174]
When using R to connect DB2 using ODBC, I get the following error only when I read the character variables. For numeric variables it works just fine.

 library(DBI)
> con<-dbConnect(odbc::odbc(),DSN="xxxx",UID,PWD)
> dbGetQuery(con, "select cust_key from customer fetch first 5 rows only")
     CUST_KEY
1 3022015993
2 3022016011
3 3022016045
4 3022016091
5 3022016134
> dbGetQuery(con, "select first_nm from customer fetch first 5 rows only")
Error in new_result(connection@ptr, statement) : std::bad_alloc

any thought on why this may be the case? I would really appreciate any insights.

**Comment by jimhester** [comment]
I am not sure and do not have a DB2 instance to attempt to reproduce this.

**Comment by wphampton** [comment]
@mbfsdatascience did you ever resolve this? I have the same issue using SyBase ASE.

**Comment by wphampton** [comment]
My problem turned out to be totally unrelated to this package.

The solution for me was to make sure I was using an ODBC driver that supports an 8-bytes SQLLEN datatype. On Linux running the command `odbcinst -j` showed me that the SQLLEN Size was 8 for my installed unixODBC version 2.3.1 running on RHEL Server 7.5 64-bit. By default the Sybase driver was symlinked to the 4-bytes version. Following the instructions below I made sure I was linking to the 8-bytes version and my problems were solved. Oddly I never experienced these issues when issuing commands via `isql` but...<shrug?> Perhaps this will work for @mbfsdatascience with DB2.

http://infocenter.sybase.com/help/topic/com.sybase.infocenter.dc20155.1600/doc/html/san1361915533650.html
